kasa/device_factory.py

```diff
@@ -180,19 +180,23 @@ def get_protocol(
     config: DeviceConfig,
 ) -> BaseProtocol | None:
     """Return the protocol from the connection name."""
-    protocol_name = config.connection_type.device_family.value.split(".")[0]
     ctype = config.connection_type
+    protocol_name = ctype.device_family.value.split(".")[0]
+
+    if ctype.device_family is DeviceFamily.SmartIpCamera:
+        return SmartCamProtocol(transport=SslAesTransport(config=config))
+
+    if ctype.device_family is DeviceFamily.IotIpCamera:
+        return IotProtocol(transport=LinkieTransportV2(config=config))
+
+    if ctype.device_family is DeviceFamily.SmartTapoRobovac:
+        return SmartProtocol(transport=SslTransport(config=config))
```

Comment:
So I'm not sure what's the best way to handle this (extract into a separate method?), but I'd add least add a quick comment that this does some special handling, especially as linkiev1 might need revisiting at some point.

Reply:
It's not really an exception it's just that the mapping is simpler for these device types. Docstring updated to explain.